### DISCUSSION TITLE
Wait until master build passes before merging

### DISCUSSION THREAD

**Discussion by JakeGinnivan** [Mergifyio/mergify#2408] [Q&A] [answered]
Hi,

I am not super clear on the strict mode behavior. Does it wait for the master build to go green (so serialise the builds on master) or does it simply update the next PR, wait for it to go green and merge (without checking the build status for master).

The scenario I am struggling with is master does a deployment, so I want to serialise the builds on master. The simplest way is to only merge the next PR once master goes green.

Is this scenario supported?

**Comment by sileht** [comment, marked as answer]
Hi,

The `merge` action `update the next PR, wait for it to go green and merge.` Its goal is to never merge a broken pull request.

Depending on what you try to achieve, you also have the `queue` action that allows configuring a different set of conditions for moving the pull request into the merge queue and for doing the final merge.


In the future, we may provide something to pause the merge queue if needed, maybe via API.

**Comment by jd** [comment]
IIUC, you want to wait for a CI to finish on `master` before merging the next PR?

That does not sound like something possible currently with Mergify nor GitHub. There's no link between a CI reporting on master and pull requests.

The only way to do that would have the build on `master` to report on all PR as a status check. It could report `failure` to prevent merging while it builds, and report a `success` when `master` is ready to get a new PR merged.
